library/src/main/java/org/xmtp/android/library/KeyUtil.kt

```diff
@@ -82,11 +81,22 @@ object KeyUtil {
     }
 
     fun validateConsentSignature(signature: String, clientAddress: String, peerAddress: String, timestamp: Long): Boolean {
-        val messageData = Signature.newBuilder().build().consentProofText(peerAddress, timestamp).toByteArray()
-        val signatureData = Numeric.hexStringToByteArray(signature)
-        val sig = Signature.parseFrom(signatureData)
-        val recoveredPublicKey = recoverPublicKeyKeccak256(sig.rawData.toByteString().toByteArray(), Util.keccak256(messageData))
-            ?: return false
-        return clientAddress == publicKeyToAddress(recoveredPublicKey)
+        if (timestamp > System.currentTimeMillis()) {
```

Comment:
Everywhere else in the code we use
```suggestion
        if (timestamp > Date().time) {
```